### PR TITLE
Replace dockerfiles with containerfiles

### PR DESCRIPTION
**Pull request by divyansh42**
Signed-off-by: divyansh42 <diagrawa@redhat.com>

### Related Issue(s)
Close: #57 

### Checklist

- [x] This PR includes a documentation change
- [ ] This PR does not need a documentation change
---
- [x] This PR includes test changes
- [ ] This PR's changes are already tested
---
- [ ] This change is a patch change
- [x] This change is a minor change
- [ ] This change is a major (breaking) change

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,14 +21,14 @@ After building your image, use [push-to-registry](https://github.com/redhat-acti
...
 
 | Input Name | Description | Default |
 | ---------- | ----------- | ------- |
 | arch | Label the image with this architecture, instead of defaulting to the host architecture. Refer to [Multi arch builds](#multi-arch-builds) for more information. | None (host architecture)
-| build-args | Build arguments to pass to the Docker build using `--build-arg`, if using a Dockerfile that requires ARGs. Use the form `arg_name=arg_value`, and separate arguments with newlines. | None
+| build-args | Build arguments to pass to the Docker build using `--build-arg`, if using a Containerfile that requires ARGs. Use the form `arg_name=arg_value`, and separate arguments with newlines. | None
 | context | Path to directory to use as the build context. | `.`
-| dockerfiles | The list of Dockerfile paths to perform a build using docker instructions. This is a multiline input to allow multiple Dockerfiles. | **Must be provided**
+| containerfiles | The list of Containerfile paths to perform a build using docker instructions. This is a multiline input to allow multiple Containerfiles. | **Must be provided**
```
Comment (tetchel) [inline README.md:31]: Add a note here to mention this was previously `dockerfiles` and that is now a supported alias.

File: src/utils.ts [hunk 2]
```diff
@@ -63,3 +64,48 @@ export async function findFuseOverlayfsPath(): Promise<string | undefined> {
...
+
+export function getContainerfiles(): string[] {
+    // 'containerfile' should be used over 'dockerfile',
+    // see https://github.com/redhat-actions/buildah-build/issues/57
+    const containerfiles = getInputList(Inputs.CONTAINERFILES);
+    const dockerfiles = getInputList(Inputs.DOCKERFILES);
+
+    if (containerfiles.length !== 0 && dockerfiles.length !== 0) {
+        core.warning(
+            `Please use only one input of "${Inputs.CONTAINERFILES}" and "${Inputs.DOCKERFILES}". `
```
Comment (tetchel) [inline src/utils.ts:91]: Both "containerfiles" and "dockerfiles" inputs are set. Please use only one of these two inputs, as they are aliases of one another. "containerfiles" takes precedence.

File: src/utils.ts [hunk 3]
```diff
@@ -63,3 +64,48 @@ export async function findFuseOverlayfsPath(): Promise<string | undefined> {
...
 }
+
+export function getArch(): string {
+    // 'arch' should be used over 'archs', see https://github.com/redhat-actions/buildah-build/issues/60
+    const archs = core.getInput(Inputs.ARCHS);
+    const arch = core.getInput(Inputs.ARCH);
+
+    if (arch && archs) {
+        core.warning(
+            `Please use only one input of "${Inputs.ARCH}" and "${Inputs.ARCHS}". "${Inputs.ARCH}" takes precedence, `
```
Comment (tetchel) [inline src/utils.ts:75]: update this message to be same as other one but wiih the input names changed.

File: src/utils.ts [hunk 4]
```diff
@@ -63,3 +64,48 @@ export async function findFuseOverlayfsPath(): Promise<string | undefined> {
...
+    return containerfiles.length !== 0 ? containerfiles : dockerfiles;
+}
+
+export function getInputList(name: string): string[] {
+    const items = core.getInput(name);
+    if (!items) {
+        return [];
+    }
+    return items
+        .split(/\r?\n/)
```
Comment (tetchel) [inline src/utils.ts:105]: you could use splitByNewline function for this

File: src/utils.ts [hunk 5]
```diff
@@ -63,3 +64,48 @@ export async function findFuseOverlayfsPath(): Promise<string | undefined> {
...
+        .filter((x) => x)
+        .reduce<string[]>(
+            (acc, line) => acc.concat(line).map((pat) => pat.trim()),
```
Comment (tetchel) [inline src/utils.ts:108]: the variable should probably not be called "pat"

File: README.md [hunk 6]
```diff
@@ -21,23 +21,24 @@ After building your image, use [push-to-registry](https://github.com/redhat-acti
...
-| dockerfiles | The list of Dockerfile paths to perform a build using docker instructions. This is a multiline input to allow multiple Dockerfiles. | **Must be provided**
+| containerfiles* | The list of Containerfile paths to perform a build using docker instructions. This is a multiline input to allow multiple Containerfiles. | **Must be provided**
 | extra-args | Extra args to be passed to buildah bud. Separate arguments by newline. Do not use quotes. | None
 | image | Name to give to the output image. | **Must be provided**
 | layers | Set to true to cache intermediate layers during the build process. | None
 | oci | Build the image using the OCI format, instead of the Docker format. By default, this is `false`, because images built using the OCI format have issues when published to Dockerhub. | `false`
 | tags | The tags of the image to build. For multiple tags, separate by a space. For example, `latest ${{ github.sha }}` | `latest`
+> *Input `containerfiles` was previously `dockerfiles`. Now input `dockerfiles` is supported alias of `containerfiles`. For details see the [linked issue](https://github.com/redhat-actions/buildah-build/issues/57).
```
Comment (tetchel) [inline README.md:37]: *The `containerfiles` input was previously `dockerfiles`. Now `dockerfiles` is an alias for `containerfiles`. For details see [the issue](https://github.com/redhat-actions/buildah-build/issues/57)

File: src/utils.ts [hunk 7]
```diff
@@ -63,3 +64,50 @@ export async function findFuseOverlayfsPath(): Promise<string | undefined> {
...
+}
+
+export function getInputList(name: string): string[] {
+    const items = core.getInput(name);
+    if (!items) {
+        return [];
+    }
+    const splitItems = splitByNewline(items);
+    return splitItems
+        .filter((x) => x)
```
Comment (tetchel) [inline src/utils.ts:108]: this filter does not do anything

Comment (divyansh42) [reply src/utils.ts:108]: yeah, I was also thinking the same.
But avoided removing, as there might be any case which I might be not aware of.

Comment (divyansh42) [reply src/utils.ts:108]: Removed!

**Review by tetchel — APPROVED** [review]

**Pull request merged**